### ISSUE TITLE
More detailed "How to"

### ISSUE THREAD

**Issue by Brawmario** [Brawmario/libsm64-godot#9]

**Comment by RegaronDev** [comment]
in Godot 4.4 mario's texture just corrupts 
How could I fix this?

![Image](https://github.com/user-attachments/assets/cfc73a10-0561-4939-9896-47954c1cf62f)

**Comment by Brawmario** [comment]
I'll open a proper issue.

**Comment by Brawmario** [comment]
Opened #32. Next time please use the proper channel for this.

**Comment by RegaronDev** [comment]
Ok I dont want to bother you that much (hope not bothering at all), but is it possible to drag enemies such as goombas in libsm64?
Since I dont really want to script one my own, before wanted to ask if there is a way in so. thx for reading

**Comment by Brawmario** [comment]
No, `libsm64` doesn't have support exposing internal SM64 entities other than Mario himself. There's no problem in asking.

**Comment by RegaronDev** [comment]
How can I let Mario jump off edges and fall?

**Comment by RegaronDev** [comment]
Hellooo, How can I let Mario fall from the ground? like an empty space? thats all I am asking please

**Comment by ARP603** [comment]
> Hellooo, How can I let Mario fall from the ground? like an empty space? thats all I am asking please

Hi. Do you mean like falling to the void? As in, Mario falls from the map and dies with the "waaaa" sound?
If that's what you're trying to accomplish, I don't know if there's an easy way to do it with the exposed methods from LibSM64, but maybe you could try to simulate the effect by creating an invisible large surface below the "ground" (with a considerable distance in between) and detect if the player reaches a certain Y level before hitting the invisible surface. You could then trigger a death event with the sound effect or whatever you want to add.

**Comment by ARP603** [comment]
I don't really know a lot about SM64, libsm64 or Godot itself, but I'm messing around with the stuff just for fun. I would also like some guidance, if there's someone able to help me:

I'm trying to load Mario in a scene with the model "Peach's Castle Exterior" from _The Models Resource_. To fix some issues with textures, I used Blender to export the model as a ".glb" file. To access the meshes from the resulting scene, I ticked the "Editable Children" box and then I added the meshes that I cared about to the static surfaces group. However, I end up getting the following error (without any other messages):

![Image](https://github.com/user-attachments/assets/2dc0283e-8e26-42f4-ba90-8cfba17b9888)

I'm able to load Mario over basic meshes from Blender using the same method and file format. My interpretation is that something about the model itself is not right, so what is the criteria used to load meshes as static surfaces?

**Comment by Brawmario** [comment]
> I don't really know a lot about SM64, libsm64 or Godot itself, but I'm messing around with the stuff just for fun. I would also like some guidance, if there's someone able to help me:
> 
> I'm trying to load Mario in a scene with the model "Peach's Castle Exterior" from _The Models Resource_. To fix some issues with textures, I used Blender to export the model as a ".glb" file. To access the meshes from the resulting scene, I ticked the "Editable Children" box and then I added the meshes that I cared about to the static surfaces group. However, I end up getting the following error (without any other messages):
> 
> ![Image](https://github.com/user-attachments/assets/2dc0283e-8e26-42f4-ba90-8cfba17b9888)
> 
> I'm able to load Mario over basic meshes from Blender using the same method and file format. My interpretation is that something about the model itself is not right, so what is the criteria used to load meshes as static surfaces?

Hello, could you share your project or just a minimal version of it? Maybe just a screenshot of the scene should be enough. Any mesh should work, as long as it's triangles aren't too small (but considering that you're using an actual model from SM64 this shouldn't be the case.

**Comment by ARP603** [comment]
Thanks for the response!

This is Mario:

![Image](https://github.com/user-attachments/assets/516b689a-997e-4c98-94da-887222f33459)

and this is the specific mesh that I've added to the surfaces group:

![Image](https://github.com/user-attachments/assets/8b972699-df2f-4f35-a452-8432c280b9a1)

When importing the model, the whole scene is way too small so I changed its scale in Godot. Could that have something to do with what you said?

**Comment by Brawmario** [comment]
Hmm, the scaling should work normally. Try creating a flat plane mesh and add it to the Static Surfaces group underneath the castle mesh, big enough to span the entire underside of the model, or adding more of the ground meshes to the Static Surfaces group, LibSM64 can be a little finicky with only concave surfaces like this.

**Comment by RegaronDev** [comment]
> > Hellooo, How can I let Mario fall from the ground? like an empty space? thats all I am asking please
> 
> Hi. Do you mean like falling to the void? As in, Mario falls from the map and dies with the "waaaa" sound? If that's what you're trying to accomplish, I don't know if there's an easy way to do it with the exposed methods from LibSM64, but maybe you could try to simulate the effect by creating an invisible large surface below the "ground" (with a considerable distance in between) and detect if the player reaches a certain Y level before hitting the invisible surface. You could then trigger a death event with the sound effect or whatever you want to add.

Yea, how could i really do that? Ive been trying to do so but suffering with it 😢

**Comment by RegaronDev** [comment]
> I don't really know a lot about SM64, libsm64 or Godot itself, but I'm messing around with the stuff just for fun. I would also like some guidance, if there's someone able to help me:
> 
> I'm trying to load Mario in a scene with the model "Peach's Castle Exterior" from _The Models Resource_. To fix some issues with textures, I used Blender to export the model as a ".glb" file. To access the meshes from the resulting scene, I ticked the "Editable Children" box and then I added the meshes that I cared about to the static surfaces group. However, I end up getting the following error (without any other messages):
> 
> ![Image](https://github.com/user-attachments/assets/2dc0283e-8e26-42f4-ba90-8cfba17b9888)
> 
> I'm able to load Mario over basic meshes from Blender using the same method and file format. My interpretation is that something about the model itself is not right, so what is the criteria used to load meshes as static surfaces?

try enabling blender export which doesnt let u export .glb right away, you can even just drag the blend file to scene and everything loads up

**Comment by RegaronDev** [comment]
I tried adding a large plane with an invisible material, but Mario doesn't slip or fall into the void.
I'd really appreciate some help with this.
I'm trying to achieve the same effect as in "Infinite Mario 64" by Twilight, where Mario actually falls into the void.
All I need is for Mario to fall. Resetting his position is easy for me since I've read the script and know how to handle Mario's spawning, etc.

**Comment by WhippyWoo** [comment]
Sorry for lowkey dropping in on this, but I was wondering if there were any plans to make marios swimming more accessible? Its been rough xDD
